src/globus_cli/commands/flows/update.py

```diff
@@ -0,0 +1,210 @@
+
+            Example: Path to JSON file:
+
+            \b
+            --definition definition.json
+    """,  # noqa: E501
+)
+@click.option(
+    "--owner",
+    type=str,
+    help="""
+        Assign ownership to your Globus Auth principal ID.
+
+        This option can only be used to take ownership of a flow,
+        and your Globus Auth principal ID must already be a flow administrator.
+
+        This option cannot currently be used to assign ownership to an arbitrary user.
+    """,
+)
+@subtitle_option
+@description_option
+@input_schema_option
+# "--no-Xs" options must exist to clear lists of things (like administrators).
+# These must be mutually exclusive with the flags that set options.
+@administrators_option
+@click.option(
+    "--no-administrators",
+    is_flag=True,
+    default=False,
+    help="Remove all administrator permissions from the flow.",
+)
+@mutex_option_group(
+    "--no-administrators", MutexInfo("--administrator", "administrators")
+)
+@starters_option
+@click.option(
+    "--no-starters",
+    is_flag=True,
+    default=False,
+    help="Remove all starter permissions from the flow.",
+)
+@mutex_option_group("--no-starters", MutexInfo("--starter", "starters"))
+@viewers_option
+@click.option(
+    "--no-viewers",
+    is_flag=True,
+    default=False,
+    help="Remove all viewer permissions from the flow.",
+)
+@mutex_option_group("--no-viewers", MutexInfo("--viewer", "viewers"))
+@keywords_option
+@click.option(
+    "--no-keywords",
+    is_flag=True,
+    default=False,
+    help="Remove all keywords from the flow.",
+)
+@mutex_option_group("--no-keywords", MutexInfo("--keyword", "keywords"))
+@LoginManager.requires_login("flows")
+def update_command(
+    flow_id: uuid.UUID,
+    login_manager: LoginManager,
+    title: str | None,
+    definition: ParsedJSONData | None,
+    input_schema: ParsedJSONData | None,
+    subtitle: str | None,
+    description: str | None,
+    owner: str | None,
+    administrators: tuple[str, ...],
+    no_administrators: bool,
+    starters: tuple[str, ...],
+    no_starters: bool,
+    viewers: tuple[str, ...],
+    no_viewers: bool,
+    keywords: tuple[str, ...],
+    no_keywords: bool,
+) -> None:
+    """
+    Update a flow.
+    """
+
+    # Ensure that the definition is a JSON object (if provided)
+    definition_doc: dict[str, JsonValue] | None = None
+    if definition is not None:
+        if not isinstance(definition.data, dict):
+            raise click.UsageError("Flow definition must be a JSON object")
+        definition_doc = definition.data
+
+    # Ensure the input schema is a JSON object (if provided)
+    input_schema_doc: dict[str, JsonValue] | None = None
+    if input_schema is not None:
+        if not isinstance(input_schema.data, dict):
+            raise click.UsageError("--input-schema must be a JSON object")
+        input_schema_doc = input_schema.data
+
+    # Empty lists (like `xs = []`) will result in erasure of the option.
```

Comment:
```suggestion
    # Passing an empty role list (like `xs = []`) will result in erasure of the option.
```

Might be slightly clearer that this is referring to the behavior of the API rather than intended behavior or the CLI?

Reply:
This is resolved.